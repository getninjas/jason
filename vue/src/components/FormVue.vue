<template>
  <form class="jason" v-on:submit.prevent="buttonOnClick">
    <FormStep v-bind:key='key' v-for='(value, key, index) in data' :position='index' :formData='formData' :value='value' :stepBy='stepBy' :step='step' :onBackClick='onBackClick' v-if='index === step' />
  </form>
</template>

<script>
import FormStep from './FormStep';

export default {
  name: 'FormVue',
  components: {
    FormStep,
  },
  data() {
    return {
      data: {},
      formData: {},
      stepBy: false,
      step: 0,
      totalSteps: 0,
    }
  },
  created() {
    this.getFormInfo();
  },
  methods: {
    getFormInfo: function () {
      // fetch('https://gist.githubusercontent.com/nathpaiva/6e2bda071405e2e6711a642ff139dacf/raw/d6cf5ff96f8d3294972c5dbb24482d4d17a37da7/form-stepby')
      fetch('https://gist.githubusercontent.com/nathpaiva/6b1aad9203ab8c3abf97113c45e310ea/raw/5150fe9f8a2c9930ae02bb967408f45c5b2e1870/form-normal')
        .then((data) => data.json())
        .then((data) => {
          this.data = data._embedded;
          this.stepBy = data.type === 'stepby' ? true : false;
          this.totalSteps = Object.keys(data._embedded).length;
        })
        .catch((error) => { console.log(error) });
    },
    buttonOnClick: function() {
      if (this.step + 1 !== this.totalSteps) {
        ++this.step;
        return;
      }

      alert('submit');
    },
    onBackClick: function() {
      if (!this.step) return;
      --this.step;
      return;
    }
  }
};
</script>
