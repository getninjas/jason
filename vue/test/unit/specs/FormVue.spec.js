import Vue from 'vue';
import FormVue from '@/components/FormVue';
import fetch from 'isomorphic-fetch';

describe('FormVue.vue', () => {
  it('has a created hook', () => {
    expect(typeof FormVue.created).toBe('function');
  });

  it('sets the correct default data', () => {
    expect(typeof FormVue.data).toBe('function');
    const defaultData = FormVue.data();
    expect(defaultData.data).toEqual({});
  });

  it('check if form is redering form', (done) => {
    const Widget = Vue.extend(FormVue)
    const vm = new Widget();

    setTimeout(function(){
      vm.$mount();
      expect(vm.$el).toMatchSnapshot();
      done();
    }, 1000);
  });
});
