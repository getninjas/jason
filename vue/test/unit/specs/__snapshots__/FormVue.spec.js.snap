// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FormVue.vue check if form is redering form 1`] = `
<form
  class="jason"
>
  
  Oi
  
  <div>
    <h1>
      Step 0
    </h1>
     
    <div>
      <label
        for="Qual será o serviço?"
      >
        Qual será o serviço?
      </label>
      <br />
       
      <input
        id="Qual será o serviço?"
        name="Qual será o serviço?"
        placeholder="Qual será o serviço?"
        type="text"
      />
    </div>
    <div>
      <label
        for="Para quem será o serviço?"
      >
        Para quem será o serviço?
      </label>
      <br />
       
      <input
        id="Para quem será o serviço?"
        name="Para quem será o serviço?"
        placeholder="Para quem será o serviço?"
        type="text"
      />
    </div>
    <div>
      <label
        for="O serviço será para quantas pessoas?"
      >
        O serviço será para quantas pessoas?
      </label>
      <br />
       
      <input
        id="O serviço será para quantas pessoas?"
        name="O serviço será para quantas pessoas?"
        placeholder="O serviço será para quantas pessoas?"
        type="text"
      />
    </div>
    <div>
      <label
        for="Para quando você precisa deste serviço?"
      >
        Para quando você precisa deste serviço?
      </label>
      <br />
       
      <input
        id="Para quando você precisa deste serviço?"
        name="Para quando você precisa deste serviço?"
        placeholder="Para quando você precisa deste serviço?"
        type="text"
      />
    </div>
    <div>
      <label
        for="Informações Adicionais"
      >
        Informações Adicionais
      </label>
      <br />
       
      <input
        id="Informações Adicionais"
        name="Informações Adicionais"
        placeholder="Descreva o que você precisa"
        type="text"
      />
    </div>
     
    <!---->
     
    <button>
      Next
    </button>
  </div>
  <!---->
</form>
`;
